libsolidity/ast/AST.cpp

```diff
@@ -389,6 +391,46 @@ std::vector<std::pair<ASTPointer<IdentifierPath>, std::optional<Token>>> UsingFo
 	return ranges::zip_view(m_functionsOrLibrary, m_operators) | ranges::to<vector>;
 }
 
+void StructDefinition::insertEncodedSubtypes(std::set<std::string>& subtypes) const
```

Comment:
nit: can you rename to the following. The name `encode` in the compiler usually refers to the ABI encoding.
```suggestion
void StructDefinition::insertEIP712EncodedSubtypes(std::set<std::string>& subtypes) const
```

Similarly for the other functions (`encode*`) that was added.